core-plugins/src/main/java/co/cask/hydrator/plugin/batch/action/OracleExportAction.java

```diff
@@ -0,0 +1,235 @@
+      }
+      String out = CharStreams.toString(outBuffer);
+      //SQLPLUS command errors are not fetched from session.getStderr().
+      //Errors and output received after executing the command in SQLPlus prompt are the one that
+      //are printed on the SQL prompt
+      if (out.contains("ERROR at line")) {
+        throw new IOException(String.format("Error:executing sqlplus query %s on hostname %s with error message: %s",
+                                            config.queryToExecute, config.oracleServerHostname, out));
+      }
+      Path file = new Path(config.pathToWriteFinalOutput);
+      FileSystem fs = FileSystem.get(file.toUri(), new Configuration());
+      FSDataOutputStream outStream = fs.create(file);
+      BufferedWriter br = new BufferedWriter(new OutputStreamWriter(outStream, "UTF-8"));
+      br.write(out.replaceAll("(?m)^[\\s&&[^\\n]]+|[\\s+&&[^\\n]]+$", "")); //Remove multiline trailing spaces
+      br.close();
+      fs.close();
+    } finally {
+      connection.close();
+    }
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    if (!config.containsMacro("port")) {
+      config.validate();
+    }
+  }
+
+  private void init() {
+    String dbConnectionString = config.dbUsername + "/" + config.dbPassword + "@" + config.oracleSID;
+    String colSeparator = getColSeparator(config.format);
+    scriptContent = new StringBuilder();
+    scriptContent.append("set colsep " + "\"" + colSeparator + "\"" + "\n");
+    scriptContent.append("set linesize 10000" + "\n");
+    scriptContent.append("set newpage none" + "\n");
+    scriptContent.append("set wrap off" + "\n");
+    scriptContent.append("set pagesize 0" + "\n");
+    scriptContent.append("set heading off" + "\n");
+    scriptContent.append("spool on" + "\n");
+    scriptContent.append(config.queryToExecute + "\n");
+    scriptContent.append("spool off" + "\n");
+    scriptContent.append("exit");
+    //Set of commands to be executed in a session
+    String setPath = "export ORACLE_HOME=" + config.oracleHome + "; export ORACLE_SID=" + config.oracleSID + ";";
+    String createTempScriptToBeUsedForSpool = "echo '" + scriptContent.toString() + "' > /tmp/tmpHydrator.sql;";
+    String executeSpoolInSqlplus = config.oracleHome + "/bin" + "/sqlplus -s " + dbConnectionString +
+      " @/tmp/tmpHydrator.sql |  sed 's/\\s*" + colSeparator + "\\s*/" + colSeparator + "/g';";
+    String removeTempScriptUsedForSpool = "rm /tmp/tmpHydrator.sql";
+    execute = setPath + createTempScriptToBeUsedForSpool + executeSpoolInSqlplus + removeTempScriptUsedForSpool;
+  }
+
+  private String getColSeparator(String format) {
+    switch (format) {
+      case "csv":
+        return ",";
+      case "tsv":
+        return "  ";
+      case "psv":
+        return "|";
+      default:
+        throw new IllegalArgumentException(
+          String.format("Invalid format '%s'. Must be one of %s", format, Joiner.on(',').join(FORMATS)));
```

Comment:
Don't need to use Joiner here. can rely on ``toString`` of ``Set``. Also, after making the formats an Enum, you can use ``EnumSet.allOf``